### DISCUSSION TITLE
fzf for filtering/sorting

### DISCUSSION THREAD

**Discussion by mil-ad** [davatorium/rofi#1872] [Ideas/Feature Requests]
Are there any plans to improve, or even better, directly integrate fzf with rofi? I know this has been raised in the issues a number of times but I thought I'd ask again in case there are any updates. I'd love to use rofi more but its filtering/sorting is nowhere close to fzf.

Separately, the configuration for the fuzzy/fzf functionality that is currently supported in rofi is quite confusing. I have seen and tried a few different combination in the github issues. I'm currently doing below but I'm not sure if I'm doing it right anyway. Is there a recipe for making rofi's filtering/searching/sorting as close as possible to fzf?

```
sort: true;
sorting-method: "fzf";
case-sensitive: false;
matching: "normal";
```

**Comment by DaveDavenport** [comment]
no clue, never use fzf. If fzf works better for you, I suggest using fzf.

> Separately, the configuration for the fuzzy/fzf functionality that is currently supported in rofi is quite confusing.

Can you elaborate?

> filtering/searching/sorting as close as possible to fzf?

From what I've read before, isn't `matching: "fuzzy";` not closer to what fzf does in its matching?

**Reply by mil-ad** [reply]
> no clue, never use fzf. If fzf works better for you, I suggest using fzf.

I do use both but `fzf` runs in the terminal so they're not really comparable. The sorting/filtering algorithms however can be compared. I do have a funny shell script combined with some Xorg rules that lets me open a terminal as a window and use fzf like rofi but obviously I lose all the speed benefits of rofi.

> From what I've read before, isn't matching: "fuzzy"; not closer to what fzf does in its matching?

You'd think that but turns out no, at least according to some of the issues I've come across. Apparently `sorting-method: fzf` is what you want.

**Reply by DaveDavenport** [reply]
You keep saying about 'issues' but you DO NOT EXPLAIN what the issue is.
If you want help, be clear!  we don't know what you know, we need context. I have no *clue* what you are talking about.

I did actually spend some time and installed fzf...  For me `find | fzf +s` and   `find | rofi -dmenu -sort true -sorting-method fzf -matching fuzzy ` seems to match pretty close and the same entry turns into the same match.

**Reply by DaveDavenport** [reply]
>  Apparently sorting-method: fzf is what you want.

I was talking about matching, not sorting.

**Reply by mil-ad** [reply]
> You keep saying about 'issues' but you DO NOT EXPLAIN what the issue is.
If you want help, be clear! we don't know what you know, we need context. I have no clue what you are talking about.

Good example is [this one](https://github.com/davatorium/rofi/issues/810) but I don't I think I find your tone very positive. Looks like you're not interested in engaging in a conversation. Good luck

**Reply by DaveDavenport** [reply]
I am confused by that reply. 

* you complain about 'the issue' yet you never explain what the issue is. 
* I've ask you to elaborate and you ignore that part. 
* You reply to something without actually reading what I've said. (matching vs sorting)

What do you expect anybody to do with remarks like:
* is nowhere close to fzf
* is quite confusing. 
* they're not really comparable. 
This is not something I can use without further explanation.


Why is it wrong to ask you explain what issue you are hitting, if I do not understand it.  I tried to explain why I need that information (e.g. I don't know the context of your complain, as it is not explained).  

How is my tone relevant? I did actually put in effort to test if I can spot the problem you are referring to (but not explained).
In my opinion, you are the one that is being negative for the conversation, you complain about things not being right, but refuse to explain this. When asked to elaborate about things you highlight, you ignore this.
When I ask you to explain things, you start to complain about my tone.

The conversation did not strand with me, it is **you** that frustrates this.

**Reply by DaveDavenport** [reply]
locking..
